appcenter-data/android/src/main/java/com/microsoft/appcenter/reactnative/data/AppCenterReactNativeDataModule.java

```diff
@@ -63,27 +63,58 @@ public String getName() {
     @ReactMethod
     public void read(String documentId, String partition, ReadableMap readOptionsMap, final Promise promise) {
         ReadOptions readOptions;
-        if (readOptionsMap.hasKey(TIME_TO_LIVE_KEY)) {
+        if(readOptionsMap == null) {
+            readOptions = null;
+        } else if (readOptionsMap.hasKey(TIME_TO_LIVE_KEY)) {
             readOptions = new ReadOptions(readOptionsMap.getInt(TIME_TO_LIVE_KEY));
         } else {
             readOptions = new ReadOptions(TimeToLive.DEFAULT);
         }
-        Data.read(documentId, JsonElement.class, partition, readOptions).thenAccept(new Consumer("Read failed", promise));
+        Data.read(documentId, JsonElement.class, partition, readOptions).thenAccept(new Consumer<JsonElement>("Read failed", promise));
     }
 
     @ReactMethod
-    public void create(String documentId, ReadableMap readableMap, String partition, ReadableMap writeOptionsMap, final Promise promise) {
+    public void create(String documentId, String partition, ReadableMap documentMap, ReadableMap writeOptionsMap, final Promise promise) {
         WriteOptions writeOptions;
-        if (writeOptionsMap.hasKey(TIME_TO_LIVE_KEY)) {
+        if(writeOptionsMap == null) {
```

Comment:
Same feedback as my previous comment.

Reply:
Changed, I added [`getReadOptions`](https://github.com/microsoft/appcenter-sdk-react-native/pull/620/files#diff-52125d7be1f0c3bf1e2530f2d4a4d1d8R145) and [`getWriteOptions`](https://github.com/microsoft/appcenter-sdk-react-native/pull/620/files#diff-52125d7be1f0c3bf1e2530f2d4a4d1d8R155).